metricbeat/module/kafka/partition/partition.go

```diff
@@ -37,57 +36,60 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 // Fetch partition stats list from kafka
 func (m *MetricSet) Fetch() ([]common.MapStr, error) {
 
-	if m.client == nil {
+	if m.broker == nil {
 		config := sarama.NewConfig()
 		config.Net.DialTimeout = m.Module().Config().Timeout
 		config.Net.ReadTimeout = m.Module().Config().Timeout
 		config.ClientID = "metricbeat"
 
-		client, err := sarama.NewClient([]string{m.Host()}, config)
+		broker := sarama.NewBroker(m.Host())
+		err := broker.Open(config)
 		if err != nil {
 			return nil, err
 		}
-		m.client = client
+		m.broker = broker
 	}
 
-	topics, err := m.client.Topics()
+	response, err := m.broker.GetMetadata(&sarama.MetadataRequest{})
```

Comment:
how about optional topic names configuration? e.g. `metricset.kafka.partitions.topics=[...]` with default being empty string. Would this work here?

Question: does the call to GetMetadata return the complete cluster-state or only topics available to the broker it's ID?